homeassistant/helpers/template.py

```diff
@@ -2074,80 +2031,227 @@ def wrapper(*args, **kwargs):
+            Ext("min", glob=min_max_from_filter(self.filters["min"], "min")),
+            Ext("version", filt=version, glob=version),
+            Ext(
+                "entry_id",
+                filt=pass_context(hassfunction(entry_id)),
+                glob=hassfunction(entry_id),
+                require_hass=True,
+            ),
+            Ext(
+                "device_entities",
+                filt=pass_context(hassfunction(device_entities)),
+                glob=hassfunction(device_entities),
+                require_hass=True,
+            ),
+            Ext(
                 "closest",
+                filt=pass_context(hassfunction(closest_filter)),
+                glob=hassfunction(closest),
+                support_limited=False,
+                require_hass=True,
+            ),
+            Ext(
                 "distance",
+                glob=hassfunction(distance),
+                support_limited=False,
+                require_hass=True,
+            ),
+            Ext(
                 "expand",
+                filt=pass_context(hassfunction(expand)),
+                glob=hassfunction(expand),
+                support_limited=False,
+                require_hass=True,
+            ),
+            Ext(
+                "device_attr",
+                filt=pass_context(hassfunction(device_attr)),
+                glob=hassfunction(device_attr),
+                support_limited=False,
+                require_hass=True,
+            ),
+            Ext(
+                "is_device_attr",
+                glob=hassfunction(is_device_attr),
+                support_limited=False,
+                require_hass=True,
+            ),
+            Ext(
+                "device_id",
+                filt=pass_context(hassfunction(device_id)),
+                glob=hassfunction(device_id),
+                support_limited=False,
+                require_hass=True,
+            ),
+            Ext(
+                "area_id",
+                filt=pass_context(hassfunction(area_id)),
+                glob=hassfunction(area_id),
+                support_limited=False,
+                require_hass=True,
+            ),
+            Ext(
+                "area_name",
+                filt=pass_context(hassfunction(area_name)),
+                glob=hassfunction(area_name),
+                support_limited=False,
+                require_hass=True,
+            ),
+            Ext(
+                "area_entities",
+                filt=pass_context(hassfunction(area_entities)),
+                glob=hassfunction(area_entities),
+                require_hass=True,
+            ),
+            Ext(
+                "area_devices",
+                filt=pass_context(hassfunction(area_devices)),
+                glob=hassfunction(area_devices),
+                require_hass=True,
+            ),
+            Ext(
+                "integration_entities",
+                filt=pass_context(hassfunction(integration_entities)),
```

Comment:
Can you make the assign_func() method wrap this automatically.  It looks like filter is always wrapped for require_hass.

Reply:
Actually it's not clear to me what this is supposed to be doing.  `hassfunction` already calls pass_context on the provided function so is this actually doing anything extra?

Reply:
> Actually it's not clear to me what this is supposed to be doing. hassfunction already calls pass_context on the provided function so is this actually doing anything extra?

There's a bit of pre-existing magic happening here that I just retained from the original implementation and I don't fully understand what it was doing.  The function wrapping is clear enough from the comment (ensure that the result isn't just baked in when the environment is constructed), but there's clearly more to it.

> Can you make the assign_func() method wrap this automatically. It looks like filter is always wrapped for require_hass.

This is some of that magic.  I'm sure that I'm missing some Python nuance, but I haven't been able to just pass in the function and do the wrapping later based on a parameter because it fails to get the name of the function:

```
  File "/workspaces/core/homeassistant/helpers/template.py", line 1993, in hassfunction
    def wrapper(*args, **kwargs):
  File "/usr/local/lib/python3.9/functools.py", line 52, in update_wrapper
    value = getattr(wrapped, attr)
  File "/workspaces/core/homeassistant/helpers/template.py", line 673, in __getattr__
    raise TemplateError(f"Invalid domain name '{name}'")
homeassistant.exceptions.TemplateError: str: Invalid domain name '__name__'
```
I'll continue to noodle on it.

Reply:
Pfft, I figured it out.  Will update on the other PR.

Reply:
Haha, glad you figured it out.  Hooray for random magic!